fluid/PaddleCV/human_pose_estimation/lib/mpii_reader.py

```diff
@@ -0,0 +1,290 @@
+    if os.path.exists(TMPDIR):
+        shutil.rmtree(TMPDIR)
+        os.mkdir(TMPDIR)
+    else:
+        os.mkdir(TMPDIR)
+
+    f = open(os.path.join(TMPDIR, filename), 'w')
+    f.close()
+
+    cv2.imwrite(os.path.join(TMPDIR, 'flip.jpg'), data_numpy)
+    cv2.imwrite(os.path.join(TMPDIR, 'input.jpg'), input)
+    for i in range(NUM_JOINTS):
+        cv2.imwrite(os.path.join(TMPDIR, 'target_{}.jpg'.format(i)), cv2.applyColorMap(
+            np.uint8(np.expand_dims(target[i], 2)*255.), cv2.COLORMAP_JET))
+        cv2.circle(input, (int(joints[i, 0]), int(joints[i, 1])), 5, [170, 255, 0], -1)
+    cv2.imwrite(os.path.join(TMPDIR, 'input_kps.jpg'), input)
+
+def _generate_target(joints, joints_vis):
+    """
+    :param joints:  [num_joints, 3]
+    :param joints_vis: [num_joints, 3]
+    :return: target, target_weight(1: visible, 0: invisible)
+    """
+    target_weight = np.ones((NUM_JOINTS, 1), dtype=np.float32)
+    target_weight[:, 0] = joints_vis[:, 0]
+
+    assert TARGET_TYPE == 'gaussian', \
+        'Only support gaussian map now!'
+
+    if TARGET_TYPE == 'gaussian':
+        target = np.zeros((NUM_JOINTS,
+                           HEATMAP_SIZE[1],
+                           HEATMAP_SIZE[0]),
+                           dtype=np.float32)
+
+        tmp_size = SIGMA * 3
+
+        for joint_id in range(NUM_JOINTS):
+            feat_stride = np.array(IMAGE_SIZE) / np.array(HEATMAP_SIZE)
+            mu_x = int(joints[joint_id][0] / feat_stride[0] + 0.5)
+            mu_y = int(joints[joint_id][1] / feat_stride[1] + 0.5)
+
+            # Check that any part of the gaussian is in-bounds
+            ul = [int(mu_x - tmp_size), int(mu_y - tmp_size)]
+            br = [int(mu_x + tmp_size + 1), int(mu_y + tmp_size + 1)]
+            if ul[0] >= HEATMAP_SIZE[0] or ul[1] >= HEATMAP_SIZE[1] \
+                    or br[0] < 0 or br[1] < 0:
+                # If not, just return the image as is
+                target_weight[joint_id] = 0
+                continue
+
+            # # Generate gaussian
+            size = 2 * tmp_size + 1
+            x = np.arange(0, size, 1, np.float32)
+            y = x[:, np.newaxis]
+            x0 = y0 = size // 2
+            # The gaussian is not normalized, we want the center value to equal 1
+            g = np.exp(- ((x - x0) ** 2 + (y - y0) ** 2) / (2 * SIGMA ** 2))
+
+            # Usable gaussian range
+            g_x = max(0, -ul[0]), min(br[0], HEATMAP_SIZE[0]) - ul[0]
+            g_y = max(0, -ul[1]), min(br[1], HEATMAP_SIZE[1]) - ul[1]
+            # Image range
+            img_x = max(0, ul[0]), min(br[0], HEATMAP_SIZE[0])
+            img_y = max(0, ul[1]), min(br[1], HEATMAP_SIZE[1])
+
+            v = target_weight[joint_id]
+            if v > 0.5:
+                target[joint_id][img_y[0]:img_y[1], img_x[0]:img_x[1]] = \
+                    g[g_y[0]:g_y[1], g_x[0]:g_x[1]]
+
+    return target, target_weight
+
+def data_augmentation(sample, is_train):
```

Comment:
The data aug between mpii and coco is different?

Is there any same fucntion in mpii_reader.py and coco_reader.py? If have, please put them in an common file